tests/sklearn/test_sklearn_autolog.py

```diff
@@ -1462,3 +1462,257 @@ def test_metric_computation_handles_absent_labels():
+    import sklearn.linear_model
+
+    mlflow.sklearn.autolog()
+    from sklearn.metrics import roc_auc_score
+
+    X, y = get_iris()
+    lor_model = sklearn.linear_model.LogisticRegression(
+        solver="saga", max_iter=100, random_state=0
+    )
+    with mlflow.start_run() as run:
+        lor_model.fit(X, y)
+        y_prob = lor_model.predict_proba(X)
+        y_true_onehot = np.eye(3)[y]
+        roc_auc_metric = roc_auc_score(y_true_onehot, y_prob)
+
+    params, metrics, tags, artifacts = get_run_data(run.info.run_id)
+    assert metrics['roc_auc_score_X'] == roc_auc_metric
+
+
+def test_is_metrics_value_loggable():
+    is_metrics_value_loggable = \
+        mlflow.sklearn._autologging_metrics_manager.is_metrics_value_loggable
+    assert is_metrics_value_loggable(3)
+    assert is_metrics_value_loggable(3.5)
+    assert is_metrics_value_loggable(np.int(3))
+    assert is_metrics_value_loggable(np.float32(3.5))
+    assert not is_metrics_value_loggable(True)
+    assert not is_metrics_value_loggable(np.bool(True))
+    assert not is_metrics_value_loggable([1, 2])
+    assert not is_metrics_value_loggable(np.array([1, 2]))
+
+
+def test_nested_metric_call_is_disabled():
+    import sklearn.linear_model
+    mlflow.sklearn.autolog()
+
+    X, y = get_iris()
+    eval1_X, eval1_y = X[0::3], y[0::3]
+    lr_model = sklearn.linear_model.LinearRegression()
+
+    with mlflow.start_run():
+        # test post training metric logging disabled in fit scope
+        with mock.patch('mlflow.sklearn._AutologgingMetricsManager.log_post_training_metric') \
+                as patched_log_post_training_metric:
+            lr_model.fit(X, y)
+            patched_log_post_training_metric.assert_not_called()
+
+        # test post training metric logging called only once in model.score
+        with mock.patch('mlflow.sklearn._AutologgingMetricsManager.log_post_training_metric') \
+                as patched_log_post_training_metric:
+            lr_model.score(eval1_X, eval1_y)
+            patched_log_post_training_metric.call_count == 1 and \
+                patched_log_post_training_metric.call_args[0][1] == 'r2_score_eval1_X'
+
+        # test post training metric logging disabled in eval_and_log_metrics
+        with mock.patch('mlflow.sklearn._AutologgingMetricsManager.log_post_training_metric') \
+                as patched_log_post_training_metric:
+            mlflow.sklearn.eval_and_log_metrics(lr_model, eval1_X, eval1_y, prefix='test1')
+            patched_log_post_training_metric.assert_not_called()
+
+
+def test_multi_model_interleaved_fit_and_post_train_metric_call():
+    import sklearn.linear_model
+    mlflow.sklearn.autolog()
+    from sklearn.metrics import mean_squared_error
+
+    X, y = get_iris()
+    eval1_X, eval1_y = X[0::3], y[0::3]
+    eval2_X, eval2_y = X[1::3], y[1::3]
+
+    lr_model1 = sklearn.linear_model.LinearRegression(positive=False)
+    lr_model2 = sklearn.linear_model.LinearRegression(positive=True)
```

Comment:
How confident are we that these models will produce different results? If they're the same, the MSEs may be the same. Iris is often really easy to train on, though linear regression models should also be pretty bad...

Reply:
The two models are quite different.
`positive=True` means force model coefficient being positive.

See metric comparison:
```
model1_r2_score=0.8010861440188818
model2_r2_score=0.5895894699419326
model1_mse=0.13120357940514557
model2_mse=0.2790791604394858
```

Reply:
I change it to use `fit_intercept`, old sklearn version does not support `positive` param